flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -222,9 +235,12 @@ private void updateStatusBeforeFirstDeployment(
             CR cr, SPEC spec, Configuration deployConfig, STATUS status, KubernetesClient client) {
         if (spec.getJob() != null) {
             var initialUpgradeMode = UpgradeMode.STATELESS;
+            var snapshotRef = spec.getJob().getFlinkStateSnapshotReference();
             var initialSp = spec.getJob().getInitialSavepointPath();
 
-            if (initialSp != null) {
+            if (snapshotRef != null) {
+                initialUpgradeMode = UpgradeMode.SAVEPOINT;
+            } else if (initialSp != null) {
                 status.getJobStatus()
                         .getSavepointInfo()
                         .setLastSavepoint(Savepoint.of(initialSp, SnapshotTriggerType.UNKNOWN));
```

Comment:
shouldn't the last savepoint info be updated in case the snapshot ref is provided? This logic is there to handle failures before the initial startup so should apply to both